src/main/java/WayofTime/alchemicalWizardry/common/items/SacrificialDagger.java

```diff
@@ -58,93 +63,102 @@ public void addInformation(ItemStack stack, EntityPlayer par2EntityPlayer, List
      */
     @Override
     public void onPlayerStoppedUsing(ItemStack stack, World world, EntityPlayer player, int itemInUseCount) {
-        // if(itemInUseCount < 32)
-        // {
-        // return;
-        // }
+        int usageCount = this.getMaxItemUseDuration(stack) - itemInUseCount;
+        int hpCount = (int) Math.min(((usageCount * 0.35)), player.getMaxHealth());
 
-        PlayerSacrificeHandler.sacrificePlayerHealth(player);
-    }
+        Random rnd = new Random();
+        int shuffle = rnd.nextInt(3);
 
-    @Override
-    public int getMaxItemUseDuration(ItemStack stack) {
-        return 72000;
-    }
+        if (player instanceof FakePlayer) {
+            return;
+        }
 
-    /**
-     * returns the action that specifies what animation to play when the items is being used
-     */
-    @Override
-    public EnumAction getItemUseAction(ItemStack stack) {
-        return EnumAction.bow;
-    }
+        if (player.isPotionActive(AlchemicalWizardry.customPotionSoulFray)) {
+            if (world.isRemote) player.addChatComponentMessage(
+                    new ChatComponentTranslation("message.sacrifice.soulfray" + shuffle)
+                            .setChatStyle(new ChatStyle().setColor(EnumChatFormatting.RED)));
+            return;
+        }
 
-    @Override
-    public ItemStack onItemRightClick(ItemStack stack, World world, EntityPlayer player) {
-        if (this.canUseForSacrifice(stack)) {
-            player.setItemInUse(stack, this.getMaxItemUseDuration(stack));
-            return stack;
+        if (hpCount < 1) {
+            return;
         }
 
-        if (!player.capabilities.isCreativeMode) {
-            SacrificeKnifeUsedEvent evt = new SacrificeKnifeUsedEvent(player, true, true, 2);
+        if (this.canUseForSacrifice(stack)) {
+            PlayerSacrificeHandler.sacrificePlayerHealth(player);
+        } else {
+            SacrificeKnifeUsedEvent evt = new SacrificeKnifeUsedEvent(player, true, true, hpCount);
+
             if (MinecraftForge.EVENT_BUS.post(evt)) {
-                return stack;
+                return;
             }
 
-            if (evt.shouldDrainHealth) {
-                player.setHealth(player.getHealth() - 2);
+            if (evt.shouldDrainHealth && !player.capabilities.isCreativeMode) {
+                if (!world.isRemote && player.getHealth() <= hpCount) {
+                    DamageSourceBloodMagic damageSrc = DamageSourceBloodMagic.INSTANCE;
+                    player.hurtResistantTime = 0;
+                    player.attackEntityFrom(damageSrc, Float.MAX_VALUE);
+                }
+                player.setHealth(player.getHealth() - hpCount);
+                player.addPotionEffect(
+                        new PotionEffect(
+                                new PotionEffect(AlchemicalWizardry.customPotionSoulFray.id, (1 + hpCount * 10), 0)));
```

Comment:
up to 191 seconds for 19 hearts? No, this will significantly discourage people from using the dagger from the start, especially with how much LP you need at the start of the game just to get all of the stuff set up for buff in this pr!